### ISSUE TITLE
OpenSSL-3 incompatibility

### ISSUE THREAD

**Issue by mkszuba** [Yubico/yubico-piv-tool#315]
Attempting to link yubico-piv-tool against OpenSSL-3 (currently in beta) results in compilation errors in ykcs11.c; see the build log attached to https://bugs.gentoo.org/805914 for details. If I understand the OpenSSL-3.0.0 design document correctly, the problem lies with it now being required for provider modules to implement a well-defined entry point in order to be able to communicate with the OpenSSL core.

**Comment by opoplawski** [comment]
We're hitting this now in Fedora rawhide as well.

**Comment by qpernil** [comment]
Agreed, this should be fixed at some point. We should also remove support for OpenSSL 1.0.

**Comment by neverpanic** [comment]
That's #334, I think.

**Comment by kloczek** [comment]
Just tested that PR and look like that it fixes the issue :)

**Comment by qpernil** [comment]
Fixed by https://github.com/Yubico/yubico-piv-tool/pull/334
